packages/libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -194,6 +194,11 @@ export class DiscoveryNodeSelector implements DiscoveryNodeSelectorService {
           // This will get the client to pick new discovery providers
           // if the selected one falls behind, even if requests are succeeding
           const responseClone = response.clone()
+          const contentType = responseClone.headers.get('Content-Type')
```

Comment:
so we won't reselect if it's a json req of any kind? will this cause any scenario where we won't reselect when we should have?

Reply:
it should be json, we unpack it as json right below

Reply:
yeah this is fine - we only want to reselect if unhealthy, and since this response succeeded (`response.ok`) and doesn't have the JSON info we care about (blocknumber, etc) we can assume we're still on a healthy node